lib/irb/ruby-lex.rb

```diff
@@ -176,50 +151,26 @@ def self.ripper_lex_without_warning(code, context: nil)
     $VERBOSE = verbose
   end
 
-  def find_prev_spaces(line_index)
```

Comment:
I dropped prev_spaces feature.

paste this code and type `\nend\nend`
```ruby
if 1
    def f
        abc
        de
        fg
```
then
```ruby
irb(main):001:1* if 1
irb(main):002:2*     def f
irb(main):003:2*         abc
irb(main):004:2*         de
irb(main):005:2*         fg
irb(main):006:1*     end
irb(main):007:1* end
=> :f
```

It is changed to
```ruby
irb(main):001:1* if 1
irb(main):002:2*     def f
irb(main):003:2*         abc
irb(main):004:2*         de
irb(main):005:2*     fg
irb(main):006:1*   end
irb(main):007:0> end
=> :f
```

This feature is not tested, and I thought it was broken because of the behavior when inserting newline between `a` and `bc`. But what I saw was a combination of another bug (#515 will fix it).
If it's not acceptable to drop in this pull request, I'll try not to drop.
If it's acceptable, I want to try re-implement it after #515.

Reply:
I think as long as we ship this PR, #515, and the new implementation in the same release, it should be fine 👍  
Can you open an issue for this so we don't forget?

Reply:
Thx!